tools/api-markdown-documenter/src/api-item-transforms/configuration/Configuration.ts

```diff
@@ -44,18 +43,44 @@ export interface ApiItemTransformationConfiguration
 	readonly uriRoot: string;
 }
 
+/**
+ * System configuration for API Item transformation functionality.
+ *
+ * @privateRemarks
+ * TODO: ideally this type should not appear in the public API.
+ * Users should only need {@link ApiItemTransformationOptions}.
+ *
+ * @public
+ */
+export interface ApiItemTransformationConfiguration
+	extends ApiItemTransformationConfigurationBase,
+		ApiItemTransformations,
+		Required<DocumentationSuiteOptions>,
+		Required<LoggingConfiguration> {}
+
+/**
+ * Input options for API Item transformation APIs.
+ *
+ * @public
+ */
+export interface ApiItemTransformationOptions
+	extends ApiItemTransformationConfigurationBase,
+		Partial<ApiItemTransformations>,
+		DocumentationSuiteOptions,
+		LoggingConfiguration {}
+
```

Comment:
This is the crux of this PR. Explicit separate types. Allows the removal of _many_ usages of `Required<ApiItemTransformationConfiguration>`.